crates/ruff_linter/src/rules/airflow/rules/removal_in_3.rs

```diff
@@ -196,6 +247,20 @@ fn removed_method(checker: &mut Checker, expr: &Expr) {
             )),
             &_ => None,
         },
+        ["airflow", "datasets", ..] | ["airflow", "Dataset"] => match attr.as_str() {
+            "iter_datasets" => Some(Replacement::Name("iter_assets")),
+            "iter_dataset_aliases" => Some(Replacement::Name("iter_asset_aliases")),
+            &_ => None,
+        },
+        _ if SECRET_BACKEND_REGEX.is_match(&qualname.segments().join(".")) => match attr.as_str() {
+            "get_conn_uri" => Some(Replacement::Name("get_conn_value")),
+            "get_connections" => Some(Replacement::Name("get_connection")),
+            &_ => None,
+        },
```

Comment:
It should be possible to simplify the regex a bit (to ensure it runs less frequently?)
```suggestion
			["airflow", rest @ ..] if SECRET_BACKEND_REGEX.is_match(rest.join(".")) => match attr.as_str() {
            "get_conn_uri" => Some(Replacement::Name("get_conn_value")),
            "get_connections" => Some(Replacement::Name("get_connection")),
            &_ => None,
        },
```

This requires removing the `airflow` segment from the regex. The advantage is that we can skip over the match if the first segment isn't airflow

Reply:
Just updated it with later comment. Thanks!